waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -572,6 +543,44 @@ when defined(rlnzerokit):
     var rootValue = cast[ptr MerkleNode] (root.`ptr`)[]
     return ok(rootValue)
 
+
+proc updateValidRootQueue*(wakuRlnRelay: WakuRLNRelay, root: MerkleNode): void =
+  # updates the valid merkle root queue with the latest root, and pops the oldest one
```

Comment:
Does it always pop the oldest? or only when the capacity is reached? i.e., `AcceptableRootWindowSize`

Reply:
It will only pop elements when the capacity is reached